src/MongoDB.Driver/Encryption/LibMongoCryptControllerBase.cs

```diff
@@ -274,5 +272,145 @@ private async Task SendKmsRequestAsync(KmsRequest request, CancellationToken can
                 }
             }
         }
+
+        // nested type
+        private static class SocketHelper
+        {
+            public static void ResolvedAndConnect(Socket socket, EndPoint endPoint)
+            {
+#if NETSTANDARD1_5
```

Comment:
Also, I think we need to add one more condition: `&& !WINDOWS` since this issue is related to Linux and possibly mac

Reply:
On Linux + `NETSTANDARD1_5`, the socket doesn't support working with host, so it requires `IP`s. Otherwise, it throws this error: 

 This platform does not support connecting sockets to DNS endpoints via the instance Connect and ConnectAsync methods, due to the potential for a host name to map to multiple IP addresses and sockets becoming invalid for use after a failed connect attempt. Use the static ConnectAsync method, or provide to the instance methods the specific IPAddress desired.."

This solution is copied from https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Driver.Core/Core/Connections/TcpStreamFactory.cs#L54

Reply:
If only `NETSTANDARD1_5` exhibits this problem, maybe we should consider not supporting it rather than writing a whole pile of wrapper code for an EOL platform.